### ISSUE TITLE
--checksum does not work as expected

### ISSUE THREAD

**Issue by jin** [coursier/coursier#1131]
I'm looking at the `--checksum` flag for [rules_jvm_external](https://github.com/bazelbuild/rules_jvm_external/issues/125), and it doesn't seem to be working:

```
$ sha1sum /usr/local/google/home/jingwen/.cache/coursier/v1/https/repo1.maven.org/maven2/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar
88e3c593e9b3586e1c6177f89267da6fc6986f0c  /usr/local/google/home/jingwen/.cache/coursier/v1/https/repo1.maven.org/maven2/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar

$ cat /usr/local/google/home/jingwen/.cache/coursier/v1/https/repo1.maven.org/maven2/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar.sha1 
88e3c593e9b3586e1c6177f89267da6fc6986f0c

$ coursier fetch com.google.errorprone:error_prone_annotations:2.2.0 --checksum 88e3c593e9b3586e1c6177f89267da6fc6986f0c
mirrorConfFiles=List(MirrorConfFile(/usr/local/google/home/jingwen/.config/coursier/mirror.properties, true))
allMirrors0=List()
Resolution error: Error downloading com.google.errorprone:error_prone_annotations:2.2.0
  not found: /usr/local/google/home/jingwen/.ivy2/local/com.google.errorprone/error_prone_annotations/2.2.0/ivys/ivy.xml
  checksum not found: 
```

Am I doing this incorrectly?

**Comment by alexarchambault** [comment]
Yes, one can't pass actual checksums to `--checksum`, only checksum types (like `SHA-1` or `SHA-256`). There's no way to somehow pin checksums this way for now.

**Comment by mandrean** [comment]
That's a big shame. No way to verify an artifact wasn't tampered with in-transit then.

**Comment by jin** [comment]
@alexarchambault ah, that explains it. Thanks. 

Since it doesn't exist yet, I'll keep this issue open as a feature request for CLI checksum verification approaches.

**Comment by ghostbuster91** [comment]
Hi,

I would like to move forward with that ticket. Are there any technical obstacles to that?

Also there is a question of how to provide that feature. I thought about enriching `--checksum` set of options by allowing to explicitly provide checksum of the given type. 

```
coursier fetch artifact:2.2.0 --checksum SHA256:88e3c593e9b3586e1c6177f89267da6fc6986f0c
```

The downsides of above is that it will complicate `--checksum` handling. Another thing is that it is easy to distinguish between types of SHA by looking at their length. Which means that there is really no point in forcing users to specify type of SHA, although it should be allowed. But maybe that is small enough inconvenience that users can leave with.

Second option would be to introduce another flag like `--verify-checksum`. The donwside of that is that it pollutes set of coursier options with two very similar flags. 


I am inclining toward the first option, but would like to hear opinion of somebody who is more familiar with the overall codebase.

**Comment by sierikov** [comment]
I personally find the first solution to function exactly as I would expect it to. Maybe with guidance from the maintenant I can try to implement it.
